account_analytic_project/model/account_analytic_account.py

```diff
@@ -0,0 +1,112 @@
+# -*- encoding: utf-8 -*-
+"""Extend account.analytic.account."""
+##############################################################################
+#
+#    OpenERP, Open Source Management Solution
+#    This module copyright (C) 2015 Therp BV <http://therp.nl>
+#    All Rights Reserved
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+from openerp.osv import orm, fields
+from openerp.tools.translate import _
+
+
+class AccountAnalyticAccount(orm.Model):
+    """Extend account.analytic.account."""
+    _inherit = 'account.analytic.account'
+
+    def _has_projects(
+            self, cr, uid, ids, dummy_name, dummy_arg, context=None):
+        """Check wether any project for this analytic account."""
+        project_model = self.pool['project.project']
+        res = {}
+        for analytic_id in ids:
+            has_project = False
+            project_ids = project_model.search(
+                cr, uid, [('analytic_account_id', '=', analytic_id)],
+                context=context
+            )
+            if project_ids:
+                has_project = True
+            res[analytic_id] = has_project
```

Comment:
But theoretically the method could be called for multiple rows. And if not, the loop will be done in one iteration anyway.